plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerContentProvider.java

```diff
@@ -65,7 +65,9 @@ protected void handleConnectionChanged(IConnection connection, String property,
 				cache.flushFor((Connection) newValue);
 			}
 			refreshViewer(newValue);
-		} 
+		} else if (ConnectionProperties.PROPERTY_RESOURCE.equals(property)) {
+		    refreshViewer(newValue);
```

Comment:
@jeffmaury can you outline in a few words why this change to the explorer fixes the problem in the editor?